internal/keyaccess/verification.go

```diff
@@ -0,0 +1,108 @@
+package keyaccess
```

Comment:
moved this to its own package, since it has use outside of just cred verification